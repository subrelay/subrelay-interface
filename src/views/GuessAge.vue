<template>
  <h1>{{ title }}</h1>
  <div class="card">
    <div style="width: 400px; height: 130px; margin-top: 20px; border-style: dotted">
      <br />
      <span>Firstmame: {{ firstname }}</span> <br />
      <span>Age: {{ age }}</span> <br />
    </div>
    <br /><br />
    <label> Enter Firstname </label><br />
    <input
      type="text"
      v-model="search"
      style="font-size: 20px; border-radius: 10px"
      placeholder=" Name ..."
    />
    <br />
    <br />
    <button type="button" @click="getAge">Guess Age</button>
    <br />
    <br />
    <br />
    <input type="radio" value="pop" /> <label>Save my data</label>
  </div>
</template>

<script setup>
import { ref } from 'vue';

defineProps({
  title: String,
});
</script>

<script>
export default {
  data() {
    return {
      search: '',
      firstname: '',
      age: '',
    };
  },
  computed: {
    getAge() {
      fetch(`https://api.agify.io/?name=${this.search}`)
        .then((response) => response.json())
        .then((data) => {
          this.age = data.age;
          this.firstname = data.name;
          this.search = '';
        });
    },
  },
};
</script>
