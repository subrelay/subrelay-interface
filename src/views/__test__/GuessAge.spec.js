import { mount } from '@vue/test-utils';
import GuessAge from '@/views/GuessAge.vue';
// import { expect, test } from 'vitest';
const wrapper = mount(GuessAge, {
  props: { title: 'Guess User Age App' },
});

it('testing GuessAge component props', async () => {
  expect(wrapper.text()).toContain('Guess User Age App');
});

it('Test if data is a function', () => {
  expect(typeof GuessAge.data).toBe('function');
});

test('snapshot UI testing', () => {
  const wrapper = mount(GuessAge, {});
  expect(wrapper.text()).toMatchSnapshot();
});
