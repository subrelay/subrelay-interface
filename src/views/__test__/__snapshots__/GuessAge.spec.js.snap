// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot UI testing 1`] = `"Firstmame:  Age:   Enter Firstname Guess Age Save my data"`;
